src/binaryen-c.cpp

```diff
@@ -3511,6 +3511,37 @@ uint32_t BinaryenGetMemorySegmentByteOffset(BinaryenModuleRef module,
   Fatal() << "non-constant offsets aren't supported yet";
   return 0;
 }
+int BinaryenHasMemory(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.exists;
+}
+size_t BinaryenMemoryGetInitial(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.initial;
+}
+int BinaryenMemoryHasMax(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.hasMax();
+}
+size_t BinaryenMemoryGetMax(BinaryenModuleRef module) {
+  return ((Module*)module)->memory.max;
+}
+const char* BinaryenMemoryImportGetModule(BinaryenModuleRef module) {
+  auto& memory = ((Module*)module)->memory;
+  if (memory.imported()) {
+    return memory.module.c_str();
+  } else {
+    return "";
+  }
+}
+const char* BinaryenMemoryImportGetBase(BinaryenModuleRef module) {
+  auto& memory = ((Module*)module)->memory;
+  if (memory.imported()) {
+    return memory.base.c_str();
+  } else {
+    return "";
+  }
+}
+int BinaryenMemoryGetShared(BinaryenModuleRef module) {
```

Comment:
Here I'm not sure. Type upon construction is `uint8_t`, and looks like a bool, so perhaps
```suggestion
int BinaryenMemoryIsShared(BinaryenModuleRef module) {
```
? Hmm

Reply:
Should I use `uint8_t` for consistency with construction of this property or `int` for consistency with all the other booleans?

Reply:
(I'll stick with int for now on the return, but going ahead and changing from `Get` to `Is`)